dataland-frontend/src/components/pages/PortfolioOverview.vue

```diff
@@ -101,22 +103,34 @@ async function getPortfolios(): Promise<void> {
 }
 
 /**
- * Sets the current portfolio ID based on the route parameter or the first portfolio in the list.
- * If the portfolioName is not provided as route parameter, it is set to the first portfolio in the list.
+ * Sets the current portfolio ID based on the following priority:
+ * 1. If a portfolioId is provided (e.g. after modal), use it if valid.
+ * 2. If not, and a route parameter for portfolioName is present, use the corresponding portfolioId if valid.
+ * 3. If not, and a session-stored portfolioId exists, use it if valid.
+ * 4. If none of the above are valid, fall back to the first portfolio in the list.
  */
 function setCurrentPortfolioId(portfolioId?: string): void {
   if (portfolioNames.value.length === 0) {
     currentPortfolioId.value = undefined;
     return;
   }
 
-  if (portfolioId) {
+  if (portfolioId && portfolioNames.value.some((p) => p.portfolioId === portfolioId)) {
```

Comment:
```suggestion
  if (portfolioId && portfolioNames.value.some((portfolio) => portfolio.portfolioId === portfolioId)) {
```
[Zen of Python](https://en.wikipedia.org/wiki/Zen_of_Python), Rule 2: "Explicit is better than implicit"